app/src/main/java/com/bignerdranch/android/criminalintent/CrimeListFragment.kt

```diff
@@ -41,12 +39,24 @@ class CrimeListFragment: Fragment() {
         crimeRecyclerView =
             view.findViewById(R.id.crime_recycler_view) as RecyclerView
         crimeRecyclerView.layoutManager = LinearLayoutManager(context)
-
-        updateUI()
+        crimeRecyclerView.adapter = adapter
 
         return view
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        crimeListViewModel.crimeListLiveData.observe(
+            viewLifecycleOwner,
+            Observer { crimes ->
+                crimes?.let {
```

Comment:
Do you need to check for nullability of `crimes` in line `52`, i.e 
```
crimes?.let { ... }
```

Isn't `crimes` already `non-null`?